core/src/main/scala/pureconfig/ConfigSource.scala

```diff
@@ -0,0 +1,299 @@
+   */
+  def fromCursor(cur: FluentConfigCursor): ConfigObjectSource =
+    new ConfigObjectSource(cur.asObjectCursor.right.map(_.value.toConfig))
+}
+
+object ConfigSource {
+
+  /**
+   * A config source for the default loading process in Typesafe Config. Typesafe Config stacks
+   * `reference.conf` resources provided by libraries, application configs (by default
+   * `application.conf` in resources) and system property overrides, resolves them and merges them
+   * into a single config.
+   */
+  val default = ConfigObjectSource(ConfigFactoryWrapper.load())
+
+  /**
+   * A config source that always provides empty configs.
+   */
+  val empty = ConfigObjectSource(Right(ConfigFactory.empty))
+
+  /**
+   * A config source for Java system properties.
+   */
+  val systemProperties = ConfigObjectSource(ConfigFactoryWrapper.systemProperties())
+
+  /**
+   * A config source for the default reference config in Typesafe Config (`reference.conf`
+   * resources provided by libraries).
+   *
+   * As required by
+   * [[https://github.com/lightbend/config/blob/master/HOCON.md#conventional-configuration-files-for-jvm-apps the HOCON spec]],
+   * the default reference files are pre-emptively resolved - substitutions in the reference config
+   * aren't affected by application configs.
+   */
+  val defaultReference = ConfigObjectSource(ConfigFactoryWrapper.defaultReference())
+
+  /**
+   * A config source for the default reference config in Typesafe Config (`reference.conf`
+   * resources provided by libraries) before being resolved. This can be used as an alternative
+   * to `defaultReference` for use cases that require `reference.conf` to depend on
+   * `application.conf`.
+   */
+  val defaultReferenceUnresolved = resources("reference.conf")
+
+  /**
+   * A config source for the default application config in Typesafe Config (by default
+   * `application.conf` in resources).
+   */
+  val defaultApplication = ConfigObjectSource(ConfigFactoryWrapper.defaultApplication())
+
+  /**
+   * A config source for the default loading process in Typesafe Config with a custom application
+   * config source. Typesafe Config stacks `reference.conf` resources provided by libraries, the
+   * given file and system property overrides, resolves them and merges them into a single config.
+   *
+   * This method is provided here to support use cases that previously depended on
+   * `ConfigFactory.load(config)`. Creating a custom source by merging the layers manually is
+   * usually recommended as it makes the config priorities more transparent.
+   *
+   * @param appSource the source providing the application config
+   * @return a `ConfigObjectSource` for the default loading process in Typesafe Config with a
+   *         custom application config source.
+   */
+  def applicationConf(appSource: ConfigObjectSource): ConfigObjectSource =
+    systemProperties.withFallback(appSource).withFallback(defaultReference)
```

Comment:
This should use `defaultOverrides` instead of `systemProperties` in case `ConfigFactory` includes some other overrides in the future.

Reply:
Added `defaultOverrides` on b3937d2. I also clarified the behavior of the default sources in 3b0aaf (in particular, the fact that they follow Typesafe Config's behavior of ignoring missing files).